### ISSUE TITLE
systemd unit: @UNBOUND_ROOTKEY_FILE@ not part of `ReadWritePaths=`

### ISSUE THREAD

**Issue by paulmenzel** [NLnetLabs/unbound#475]
Using the systemd service unit from `contrib`, containing the line below

https://github.com/NLnetLabs/unbound/blob/77451862f0455c532aa9f6d30f8fb429d2f3a609/contrib/unbound.service.in#L76

and configuring the Unbound build with `--with-rootkey-file=/var/unbound/root.key`, starting the service fails:

    unbound[6269]: [1619522080] unbound[6269:0] fatal error: could not open autotrust file for writing, /var/unbound/root.key.6269-0-593e60: Read-only file system

Adding `/var/unbound/` to `ReadWritePaths=` fixes this.

The question is, how to fix it properly.

Using

    ReadWritePaths=@UNBOUND_RUN_DIR@ @UNBOUND_CHROOT_DIR@ @UNBOUND_ROOTKEY_FILE@

does not help with the temporary files, which are created, and wildcards are not possible.

The temporary files could be created in `$TMPDIR`, and the line above edited.

**Comment by paulmenzel** [comment]
https://github.com/NLnetLabs/unbound/blob/77451862f0455c532aa9f6d30f8fb429d2f3a609/validator/autotrust.c#L1195-L1201

The temporary file should not be created in the directory, where the root key is saved in, as that is not necessarily writable.

**Comment by Maryse47** [comment]
Is `/var/unbound` some custom path that contains only the key? Setting `--with-run-dir=/var/unbound` would make it work otherwise I don't see the way to catch it in generic config as it's prett non-standard. `/var/lib/unbound` would work o-o-t-b.

**Comment by ArchangeGabriel** [comment]
@paulmenzel `/var/unbound` is not a FHS valid path AFAIK. You should not use it, but instead `/var/lib/unbound` as proposed by @Maryse47, which would solve your issue.

**Comment by paulmenzel** [comment]
As long as the configure switch `--with-rootkey-file=` exists, it would be great to have it seamlessly integrated with the provided systemd service unit.

(Temporary files should be created in `$TMPDIR` in my opinion.)

**Comment by ArchangeGabriel** [comment]
Right, adding `@UNBOUND_ROOTKEY_FILE@` seems fine indeed, I was just saying that in your case there was another more correct solution not requiring changes in unbound. Temporary files are another issue (even if related because it tries to write them at the same place), but using the `$TMPDIR` for them might be a good idea, yes.

**Comment by Maryse47** [comment]
The problem with `$TMPDIR` or `/tmp` is they are usually world writable/readable so avoiding them makes sense.

**Comment by ArchangeGabriel** [comment]
But they are not when running with systemd, because we have `PrivateTmp=true`, which means that unbound gets its own `/tmp` that only it and root can access. Maybe this could be configured (or is that already the case?) so that people using systemd can decide where those temporary files should be written?

**Comment by Maryse47** [comment]
Yes, for systemd unit as provided in the repo this isn't an issue however most users/distros may use different one.
